README.md

```diff
@@ -1,25 +1,45 @@
-# fastcrypto
-Common cryptographic library used in software at Mysten Labs.
+# [fastcrypto]
+
+[![crate][crate-image]][crate-link]
+[![Docs][docs-image]][docs-link]
+![Apache2/MIT licensed][license-image]
+![Rust Version][rustc-image]
+
+fastcrypto` is a common cryptography library used by software at Mysten Labs. It is a wrapper library around several carefully selected crates with following considerations: 
+- Security: Whether the libraries are vulnerable to known attack vectors or possible misuses. 
+- Performance: Whether the crate performs cryptographic operations with speed after extensive benchmarking. This is critical for the Sui Network to be performant when signing and verifying large amounts of transactions and certificates. 
+- Determinism: Whether the signature is non-malleable.
+- Popularity: Whether the library is used by other consensus critical systems. 
+
+Furthermore, we extends the selected libraries with additional features:
+- Robust testing framework: [Wycheproof tests](https://github.com/google/wycheproof) and [prop tests](https://altsysrq.github.io/proptest-book/intro.html) are added when possible to protect against arbitrary inputs and crafted edge cases.
+- Zeroization: Sensitive private key materials are cleared from memory securely and proactively when it goes out of scope using [zeroize](https://docs.rs/zeroize/latest/zeroize/) trait.
+- Serialization: Effective and standardized serialization are required.
+
```

Comment:
This library will be continuously updated with more schemes and faster / more secure implementations based on benchmarking results, RFC updates and audit inputs.